pkg/registry/generic/etcd/etcd.go

```diff
@@ -427,6 +428,14 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 			if lastGraceful > 0 {
 				return out, nil
 			}
+			// If we are here, the registry supports grace period mechanism and
+			// we are intentionally delete gracelessly. In this case, we may
+			// enter a race with other k8s components. If other component wins
+			// the race, the object will not be found, and we should tolerate
+			// the NotFound error. See
+			// https://github.com/kubernetes/kubernetes/issues/19403 for
+			// details.
+			ignoreNotFound = true
 			// fall through and delete immediately
```

Comment:
fallthrough here is misleading, I think... it's not falling through this switch case... it's continuing below to ```if err := e.Storage.Delete(ctx, key, out); err != nil {```, etc